Inclusion of freebsd RIDs causes restore error on vertical builds

The changes in https://github.com/dotnet/installer/pull/18763 are attempting to update the SDK version that is used in the VMR. This is causing failures in the vertical builds:

```
Errors
    D:\a\_work\1\vmr\src\runtime\src\coreclr\tools\aot\ILCompiler\repro\repro.csproj error NU1101: Unable to find package Microsoft.NETCore.App.Runtime.freebsd-x64. No packages exist with this id in source(s): dotnet-eng, dotnet-libraries, dotnet-libraries-transport, dotnet-public, dotnet-tools, dotnet9-transport, net-sdk-supporting-feed, source-built-command-line-api, source-built-emsdk, source-built-transport-arcade, source-built-transport-cecil, source-built-transport-emsdk [D:\a\_work\1\vmr\src\runtime\Build.proj]
    D:\a\_work\1\vmr\src\runtime\src\coreclr\tools\aot\ILCompiler\repro\repro.csproj error NU1101: Unable to find package Microsoft.NETCore.App.Runtime.freebsd-arm64. No packages exist with this id in source(s): dotnet-eng, dotnet-libraries, dotnet-libraries-transport, dotnet-public, dotnet-tools, dotnet9-transport, net-sdk-supporting-feed, source-built-command-line-api, source-built-emsdk, source-built-transport-arcade, source-built-transport-cecil, source-built-transport-emsdk [D:\a\_work\1\vmr\src\runtime\Build.proj]
    D:\a\_work\1\vmr\src\runtime\src\coreclr\tools\aot\ILCompiler\repro\repro.csproj error NU1101: Unable to find package Microsoft.NETCore.App.Host.freebsd-x64. No packages exist with this id in source(s): dotnet-eng, dotnet-libraries, dotnet-libraries-transport, dotnet-public, dotnet-tools, dotnet9-transport, net-sdk-supporting-feed, source-built-command-line-api, source-built-emsdk, source-built-transport-arcade, source-built-transport-cecil, source-built-transport-emsdk [D:\a\_work\1\vmr\src\runtime\Build.proj]
    D:\a\_work\1\vmr\src\runtime\src\coreclr\tools\aot\ILCompiler\repro\repro.csproj error NU1101: Unable to find package Microsoft.NETCore.App.Host.freebsd-arm64. No packages exist with this id in source(s): dotnet-eng, dotnet-libraries, dotnet-libraries-transport, dotnet-public, dotnet-tools, dotnet9-transport, net-sdk-supporting-feed, source-built-command-line-api, source-built-emsdk, source-built-transport-arcade, source-built-transport-cecil, source-built-transport-emsdk [D:\a\_work\1\vmr\src\runtime\Build.proj]
    D:\a\_work\1\vmr\src\runtime\src\coreclr\tools\aot\ILCompiler\repro\repro.csproj error NU1101: Unable to find package Microsoft.AspNetCore.App.Runtime.freebsd-x64. No packages exist with this id in source(s): dotnet-eng, dotnet-libraries, dotnet-libraries-transport, dotnet-public, dotnet-tools, dotnet9-transport, net-sdk-supporting-feed, source-built-command-line-api, source-built-emsdk, source-built-transport-arcade, source-built-transport-cecil, source-built-transport-emsdk [D:\a\_work\1\vmr\src\runtime\Build.proj]
    D:\a\_work\1\vmr\src\runtime\src\coreclr\tools\aot\ILCompiler\repro\repro.csproj error NU1101: Unable to find package Microsoft.AspNetCore.App.Runtime.freebsd-arm64. No packages exist with this id in source(s): dotnet-eng, dotnet-libraries, dotnet-libraries-transport, dotnet-public, dotnet-tools, dotnet9-transport, net-sdk-supporting-feed, source-built-command-line-api, source-built-emsdk, source-built-transport-arcade, source-built-transport-cecil, source-built-transport-emsdk [D:\a\_work\1\vmr\src\runtime\Build.proj]
```

The project.assets.json file shows that the `net9.0` target framework has the Microsoft.NETCore.App.Host and Microsoft.AspNetCore.App.Runtime runtime packs listed as download dependencies for the freebsd RID. This causes an attempt to download these packages and it fails because they do not exist in the .NET engineering package feeds.

Binlog: [inner-build-runtime.zip](https://github.com/dotnet/installer/files/14393211/inner-build-runtime.zip)

This inclusion of the freebsd RID into the SDK was introduced by this change: https://github.com/dotnet/installer/pull/18493

/cc @Thefrank, @akoeplinger

--- a/src/coreclr/tools/aot/ILCompiler/repro/repro.csproj
+++ b/src/coreclr/tools/aot/ILCompiler/repro/repro.csproj
@@ -6,7 +6,9 @@
     <PlatformTarget>AnyCPU</PlatformTarget>
     <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
     <AppendRuntimeIdentifierToOutputPath>false</AppendRuntimeIdentifierToOutputPath>
-    <RuntimeIdentifiers>linux-x64;win-x64;osx-x64;freebsd-x64;freebsd-arm64</RuntimeIdentifiers>
+    <RuntimeIdentifiers>linux-x64;win-x64;osx-x64</RuntimeIdentifiers>
+    <!-- FreeBSD runtime/apphost packs aren't built in the official build so only reference the RIDs when targetting FreeBSD -->
+    <RuntimeIdentifiers Condition="'$(TargetOS)' == 'freebsd'">$(RuntimeIdentifiers);freebsd-x64;freebsd-arm64</RuntimeIdentifiers>
     <Configurations>Debug;Release;Checked</Configurations>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <RunAnalyzers>false</RunAnalyzers>

--- a/src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.Crossgen2.sfxproj
+++ b/src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.Crossgen2.sfxproj
@@ -12,7 +12,9 @@
     <ArchiveName>dotnet-crossgen2</ArchiveName>
     <SharedFrameworkHostFileNameOverride>crossgen2</SharedFrameworkHostFileNameOverride>
     <!-- Build this pack for any RID if building from source. Otherwise, only build select RIDs. -->
-    <RuntimeIdentifiers Condition="'$(DotNetBuildSourceOnly)' != 'true'">linux-x64;linux-musl-x64;linux-arm;linux-musl-arm;linux-arm64;linux-musl-arm64;freebsd-x64;freebsd-arm64;osx-x64;osx-arm64;win-x64;win-x86;win-arm64</RuntimeIdentifiers>
+    <RuntimeIdentifiers Condition="'$(DotNetBuildSourceOnly)' != 'true'">linux-x64;linux-musl-x64;linux-arm;linux-musl-arm;linux-arm64;linux-musl-arm64;osx-x64;osx-arm64;win-x64;win-x86;win-arm64</RuntimeIdentifiers>
+    <!-- FreeBSD runtime/apphost packs aren't built in the official build so only reference the RIDs when targetting FreeBSD -->
+    <RuntimeIdentifiers Condition="'$(DotNetBuildSourceOnly)' != 'true' and '$(TargetOS)' == 'freebsd'">$(RuntimeIdentifiers);freebsd-x64;freebsd-arm64</RuntimeIdentifiers>
     <GenerateInstallers>false</GenerateInstallers>
     <HostJsonTargetPath>tools/</HostJsonTargetPath>
     <PermitDllAndExeFilesLackingFileVersion>true</PermitDllAndExeFilesLackingFileVersion>

